src/renderer/dx/DxRenderer.cpp

```diff
@@ -12,6 +12,177 @@
 #include "../../inc/DefaultSettings.h"
 #include <VersionHelpers.h>
 
+#include <DirectXMath.h>
+#include <d3dcompiler.h>
+#include <DirectXColors.h>
+
+#pragma comment(lib, "d3dcompiler.lib")
+using namespace DirectX;
+
+
+// Quad where we draw the terminal.
+// pos is world space coordinates where origin is at the center of screen.
+// tex is texel coordinates where origin is top left.
+// Layout the quad as a triangle strip where the _screenQuadVertices are place like so.
+// 2 0
+// 3 1
+struct ShaderInput
+{
+    XMFLOAT3 pos;
+    XMFLOAT2 tex;
+} const _screenQuadVertices[] =
+{
+    { XMFLOAT3(1.f, 1.f, 0.f), XMFLOAT2(1.f, 0.f) },
+    { XMFLOAT3(1.f, -1.f, 0.f), XMFLOAT2(1.f, 1.f) },
+    { XMFLOAT3(-1.f, 1.f, 0.f), XMFLOAT2(0.f, 0.f) },
+    { XMFLOAT3(-1.f, -1.f, 0.f), XMFLOAT2(0.f, 1.f) },
+};
+
+D3D11_INPUT_ELEMENT_DESC _shaderInputLayout[] =
+{
+    { "POSITION", 0, DXGI_FORMAT_R32G32B32_FLOAT, 0, 0, D3D11_INPUT_PER_VERTEX_DATA, 0 },
+    { "TEXCOORD", 0, DXGI_FORMAT_R32G32_FLOAT, 0, D3D11_APPEND_ALIGNED_ELEMENT, D3D11_INPUT_PER_VERTEX_DATA, 0 }
+};
+
+const char _vertexShaderString[] = R"(
+struct VS_OUTPUT
+{
+    float4 pos : SV_POSITION;
+    float2 tex : TEXCOORD;
+};
+VS_OUTPUT main(float4 pos : POSITION, float2 tex : TEXCOORD)
+{
+    VS_OUTPUT output;
+    output.pos = pos;
+    output.tex = tex;
+    return output;
+}
+)";
+
+const char _pixelShaderString[] = R"(

```

Comment:
I haven't done shaders before, in earnest, but isn't there a resource file type that shader code can be placed into and loaded instead of writing it as giant const strings in the source?

Reply:
I can move this to an .hlsl file that's built by visual studio. But how do I make sure the output is packaged with application so the application can find the compiled shader output at runtime?

Reply:
Does it not carry it automatically? I would think that adding a supported compilation thing should be carried through into the binaries and into the package. We might have to page @DHowett-MSFT if it isn't automatic.

Reply:
Nope
```
C:\Users\changyl\Documents\GitHub\terminal\src\renderer\dx\DxRenderer.cpp(211)\TerminalControl.dll!00007FF87A823678: (caller: 00007FF87A822E1C) ReturnHr(4) tid(3150) 80070002 The system cannot find the file specified.
    [Microsoft::Console::Render::DxEngine::_SetupTerminalEffects(D3DReadFileToBlob(L"ScreenVertexShader.cso", &vertexBlob))]
```

Reply:
Dustin said that the only way to package it into the UWP app right now is to put the hlsl files into another project. That's going to be kind of messy so he said he's OK with leaving it in the source code for now and fixing it later. What do you think?

Reply:
Yea I think I'm okay with that, but let's compromise by defining the strings in some other header file and including them here. Does that sound reasonable?